ibmpisession/ibmpowersession.go

```diff
@@ -76,16 +99,60 @@ region : Obtained from the terraform template. Every template /resource will be
 timeout:
 useraccount:
 */
+// - deprecated: New function can be used, but is slated to become `obsolete`, Instead try using NewSession function.
 func New(iamtoken, region string, debug bool, timeout time.Duration, useraccount string, zone string) (*IBMPISession, error) {
+	fmt.Println("Warning - New function is slated to become `obsolete`, Instead try using NewSession function.")
 	session := &IBMPISession{
 		IAMToken:    iamtoken,
 		UserAccount: useraccount,
 		Region:      region,
 		Zone:        zone,
 	}
 
+	session, err := process(session, debug, timeout)
+	if err != nil {
+		return nil, err
+	}
+	return session, nil
+}
+
+/*
+The method takes in the following params
+options : This is the PIOptions param passed by clients.
+This method will create a IBMPISession based on the options passed by client.
+It will pass the newly created IBMPISession to newSubProcess method which will evaluate and set rest of the required options from session.
+*/
+func NewSession(options *PIOptions) (*IBMPISession, error) {
+	t := timeout
+	if options.Timeout.Seconds() != 0 {
+		t = options.Timeout
+	}
+
+	newSession := &IBMPISession{
```

Comment:
You are not using `IAMToken` in the new struct. Deprecating the `New` function should not mean that we won't allow auth via direct token.
This is currently in use by the Terraform provider code which is the main use case for this project.

Reply:
`NewSession` this function itself can generate that token you are talking about, user no need to generate the token in advance. Even tf can start using this function eventually.